### ISSUE TITLE
Subtractors Can Return None Which Repeatedly Crashes watchman-subprocess.py

### ISSUE THREAD

**Issue by eviljoel** [park-bench/watchman#20]
This is hard to reproduce but I've observed watchman-subprocess.py repeatedly crashing due to the following error:

```
2023-12-14 03:17:24,968 [CRITICAL] __main__: Fatal TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'
Traceback (most recent call last):
  File "/usr/share/watchman/watchman-subprocess.py", line 544, in <module>
    watchman_subprocess.start_loop()
  File "/usr/share/watchman/watchman-subprocess.py", line 114, in start_loop
    self._calculate_absolute_difference_mean_total(current_frame, last_frame)
  File "/usr/share/watchman/watchman-subprocess.py", line 229, in _calculate_absolute_difference_mean_total
    difference_image = last_frame['subtracted_image'] - current_frame['subtracted_image']
TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'
```

This completely prevents the program from working correctly.  Either restarting the daemon or re-seating the camera's USB connection corrected the issue. (I cannot remember which one fixed it.) 

Apparently `self.subtractor.apply(image)` is returning None. I'm not sure there is anything we can do to prevent the subtractor from returning None once this problems starts. We should consider notifying the user via e-mail when this occurs. Care must be taken to ensure the user isn't flooded with e-mails about this.

**Comment by eviljoel** [comment]
The error just occurred again. Here is an excerpt from the subprocess log:
```
2023-12-14 04:31:55,124 [INFO] root: Option still_running_email_subject: Encrypted message from your server.
2023-12-14 04:31:55,124 [DEBUG] root: Verifying numeric option still_running_email_max_delay.
2023-12-14 04:31:55,124 [INFO] root: Option still_running_email_max_delay: 7
2023-12-14 04:31:55,124 [DEBUG] root: Checking boundaries.
2023-12-14 04:31:55,125 [DEBUG] root: Verifying numeric option replacement_subtractor_creation_threshold.
2023-12-14 04:31:55,125 [INFO] root: Option replacement_subtractor_creation_threshold: 40
2023-12-14 04:31:55,125 [DEBUG] root: Checking boundaries.
2023-12-14 04:31:55,128 [TRACE] __main__: abs_diff_mean_total: 0.1291341146
2023-12-14 04:31:55,252 [TRACE] __main__: abs_diff_mean_total: 0.0860677083
2023-12-14 04:31:55,270 [CRITICAL] __main__: Fatal TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'
Traceback (most recent call last):
  File "/usr/share/watchman/watchman-subprocess.py", line 544, in <module>
    watchman_subprocess.start_loop()
  File "/usr/share/watchman/watchman-subprocess.py", line 114, in start_loop
    self._calculate_absolute_difference_mean_total(current_frame, last_frame)
  File "/usr/share/watchman/watchman-subprocess.py", line 229, in _calculate_absolute_difference_mean_total
    difference_image = last_frame['subtracted_image'] - current_frame['subtracted_image']
TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'

2023-12-14 04:31:55,379 [TRACE] __main__: abs_diff_mean_total: 0.0658984375
2023-12-14 04:31:55,508 [TRACE] __main__: abs_diff_mean_total: 0.1328841146
2023-12-14 04:31:55,632 [TRACE] __main__: abs_diff_mean_total: 0.0934440104
2023-12-14 04:31:55,760 [TRACE] __main__: abs_diff_mean_total: 0.1815722656
2023-12-14 04:31:55,884 [TRACE] __main__: abs_diff_mean_total: 0.1344368490
2023-12-14 04:31:56,012 [TRACE] __main__: abs_diff_mean_total: 0.1708561198
2023-12-14 04:31:56,140 [TRACE] __main__: abs_diff_mean_total: 0.1015039063
2023-12-14 04:31:56,221 [INFO] root: Validating watchman configuration.
2023-12-14 04:31:56,222 [DEBUG] root: Verifying integer option video_device_number.
2023-12-14 04:31:56,222 [INFO] root: Option video_device_number: 0
2023-12-14 04:31:56,223 [DEBUG] root: Checking boundaries.
2023-12-14 04:31:56,223 [DEBUG] root: Verifying integer option initial_frame_skip_count.
2023-12-14 04:31:56,223 [INFO] root: Option initial_frame_skip_count: 1000
``` 

This somewhat suggests there are two watchman subprocesses running at the same time.

**Comment by eviljoel** [comment]
Confirmed that unplugging the camera for a while and then plugging it back in returns watchman to normal operation.

Actually, I figured out how to reproduce the problem. Restarting watchman (via systemctl restart) with the camera plugged in seems to pretty consistently cause this error to occur. Also confirmed that there are two watchman subprocesses running at the same time.

This is a big enough issue to require a fix before the 0.8 release.
